weave-js/src/react.tsx

```diff
@@ -481,7 +481,8 @@ export const makeCallAction = (
   rootArgs: {[key: string]: any},
   rootType: Type,
   refreshAll: () => Promise<void>,
-  handleRootUpdate?: (newRoot: Node) => void
+  handleRootUpdate?: (newRoot: Node) => void,
+  ignoreNullResult: boolean = true
```

Comment:
Seems like a smell... can you remind me why this is necessary?

Reply:
because `final` is `null` when we reset the domain to empty. before this we would drop that update but with this we uptake the null

Reply:
hmmm, so there are two cases 'clientRef' and 'serverRef'. If `serverRef`, then we require string output. If `clientRef`, the resulting root is the return of the mutation, so `null` feels valid. I'm wondering if we even need the `final == null` check at all? And if not, then we can drop this `ignoreNullResult` param